src/views/virtualmachinesinstance/details/tabs/console/components/vnc-console/VncConsole.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from 'react';
+
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import RFBCreate from '@novnc/novnc/core/rfb';
+import { initLogging } from '@novnc/novnc/core/util/logging';
+import {
+  Button,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  Spinner,
+} from '@patternfly/react-core';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/Consoles/VncConsole';
+
+import { VncConsoleProps, VNCState } from './utils';
+import VncConsoleActions from './VncConsoleActions';
+
+import '@patternfly/react-styles/css/components/Consoles/VncConsole.css';
+
+const { connected, connecting, disconnected } = VNCState;
+
+export const VncConsole: React.FunctionComponent<VncConsoleProps> = ({
+  children,
+  host,
+  port = '80',
+  path = '',
+  encrypt = false,
+  resizeSession = true,
+  scaleViewport = false,
+  viewOnly = false,
+  shared = false,
+  credentials,
+  repeaterID = '',
+  vncLogging = 'warn',
+  consoleContainerId,
+  additionalButtons = [] as React.ReactNode[],
+  onDisconnected = () => null,
+  onInitFailed,
+  onSecurityFailure,
+  textConnect,
+  textConnecting,
+  textDisconnected,
+  textDisconnect,
+  textSendShortcut,
+  textCtrlAltDel,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [rfb, setRfb] = React.useState<any>();
+  const [status, setStatus] = React.useState<VNCState>(disconnected);
+  const staticRenderLocaitonRef = React.useRef(null);
+  const StaticRenderLocaiton = React.useMemo(
+    () => <div id={consoleContainerId} ref={staticRenderLocaitonRef} />,
+    [staticRenderLocaitonRef, consoleContainerId],
+  );
+
+  const url = React.useMemo(
+    () => `${encrypt ? 'wss' : 'ws'}://${host}:${port}/${path}`,
```

Comment:
maybe put `ws` and `wss` as consts in `VirtualMachinesInstancePageConsoleTabConsts.ts`?